staging/cse/windows/debug/collect-windows-logs.ps1

```diff
@@ -132,22 +132,58 @@ if ($res) {
   Write-Host "Collecting Containerd running containers - tasks"
   & ctr.exe -n k8s.io t ls > "$ENV:TEMP\$timeStamp-containerd-tasks.txt"
   $paths += "$ENV:TEMP\$timeStamp-containerd-tasks.txt"
+
+  Write-Host "Collecting Containerd running containers - snapshot"
+  & ctr.exe -n k8s.io snapshot ls > "$ENV:TEMP\$timeStamp-containerd-snapshot.txt"
+  $paths += "$ENV:TEMP\$timeStamp-containerd-snapshot.txt"
 }
 else {
   Write-Host "ctr.exe command not available"
 }
 
-# log containers the CRI plugin is aware of, and their state.
-Write-Host "Collecting CRI plugin containers"
+# log containers, pods and images the CRI plugin is aware of, and their state.
 $res = Get-Command crictl.exe -ErrorAction SilentlyContinue
 if ($res) {
+  Write-Host "Collecting CRI plugin containers"
   & crictl.exe ps -a > "$ENV:TEMP\$timeStamp-cri-containerd-containers.txt"
   $paths += "$ENV:TEMP\$timeStamp-cri-containerd-containers.txt"
+
+  Write-Host "Collecting CRI plugin pods"
+  & crictl.exe pods > "$ENV:TEMP\$timeStamp-cri-containerd-pods.txt"
+  $paths += "$ENV:TEMP\$timeStamp-cri-containerd-pods.txt"
+
+  Write-Host "Collecting CRI plugin images"
+  & crictl.exe images > "$ENV:TEMP\$timeStamp-cri-containerd-images.txt"
+  $paths += "$ENV:TEMP\$timeStamp-cri-containerd-images.txt"
 }
 else {
   Write-Host "crictl.exe command not available"
 }
 
+# use runhcs shim diagnostic tool 
+$res = Get-Command shimdiag.exe -ErrorAction SilentlyContinue
+if ($res) {
+  Write-Host "Collecting logs of runhcs shim diagnostic tool"
+  foreach ($line in shimdiag.exe list) {
```

Comment:
Will `shimdiag.exe list` be only executed once? It is better to run it before for loop.

Reply:
From the result, it is executed only once. I will move it to the condition before the loop.